war/src/main/webapp/scripts/hudson-behavior.js

```diff
@@ -2923,18 +2923,24 @@ function applyErrorMessage(elt, rsp) {
     if (rsp.status == 200) {
         elt.innerHTML = rsp.responseText;
     } else {
-        var id = 'valerr' + (iota++);
-        elt.innerHTML = '<a href="" onclick="document.getElementById(\'' + id
-        + '\').style.display=\'block\';return false">ERROR</a><div id="'
-        + id + '" style="display:none">' + rsp.responseText + '</div>';
-        var error = document.getElementById('error-description'); // cf. oops.jelly
-        if (error) {
-            var div = document.getElementById(id);
-            while (div.firstElementChild) {
-                div.removeChild(div.firstElementChild);
-            }
-            div.appendChild(error);
-        }
+        var parser = new DOMParser();
+        var doc = parser.parseFromString(rsp.responseText, 'text/html');
+        var message = doc.querySelector('#error-description') || doc.body;
```

Comment:
I couldn’t test this special-cased element as I don’t know how to trigger it. Any hints appreciated.
The original implementation didn’t clearly specify that the `error-description` element is in the returned part of the DOM and not anywhere else but it’s the only way the code made sense (otherwise, why inject the elements only to remove them again?), so that’s what I assumed.

Reply:
This is from `oops.jelly`; it is a hack I did a few years ago mostly that "Apply" buttons get a useful error message without including an entire layout in their popup-looking dialog.

Might still be triggerable when trying to Apply a job configuration with a syntactically invalid "Build periodically" cron pattern.